impl/src/main/java/io/kroxylicious/testing/kafka/common/ConstraintUtils.java

```diff
@@ -177,12 +178,18 @@ private static <A extends Annotation> A mkAnnotation(Class<A> annoType, Map<Stri
                 annoType.getDeclaredMethod(member);
             }
             catch (NoSuchMethodException e) {
-                throw new RuntimeException(e);
+                throw new IllegalArgumentException("Annotation %s does not have an element %s".formatted(annoType.getName(), member));
             }
         }
-        Objects.requireNonNull(members);
+        var defaults = Arrays.stream(annoType.getDeclaredMethods())
```

Comment:
Previous the proxy wasn't respecting the annotation defaults.  I had a workaround in the production code for the test shortcoming.  This change addresses the test shortcoming.